File: sdk/@launchdarkly/mobile-dotnet/android/native/LDObserve/src/main/java/com/example/LDObserve/OptionsBridge.kt
package com.launchdarkly.LDNative

public class LDObservabilityOptions {
    @JvmField var isEnabled: Boolean = true
    @JvmField var serviceName: String = ""
    @JvmField var serviceVersion: String = ""
    @JvmField var otlpEndpoint: String = ""
    @JvmField var backendUrl: String = ""
    @JvmField var contextFriendlyName: String? = null
    @JvmField var attributes: HashMap<String, Any?>? = null

    constructor()

    constructor(
        isEnabled: Boolean,
        serviceName: String,
        serviceVersion: String,
        otlpEndpoint: String,
        backendUrl: String,
        contextFriendlyName: String?,
        attributes: HashMap<String, Any?>? = null
    ) {
        this.isEnabled = isEnabled
        this.serviceName = serviceName
        this.serviceVersion = serviceVersion
        this.otlpEndpoint = otlpEndpoint
        this.backendUrl = backendUrl
        this.contextFriendlyName = contextFriendlyName
        this.attributes = attributes
    }
}

public class LDPrivacyOptions {
    @JvmField var maskTextInputs: Boolean = true
    @JvmField var maskWebViews: Boolean = false
    @JvmField var maskLabels: Boolean = false
    @JvmField var maskImages: Boolean = false
    @JvmField var minimumAlpha: Double = 0.02

    constructor()

    constructor(
        maskTextInputs: Boolean,
        maskWebViews: Boolean,
        maskLabels: Boolean,
        maskImages: Boolean,
        minimumAlpha: Double
    ) {
        this.maskTextInputs = maskTextInputs
        this.maskWebViews = maskWebViews
        this.maskLabels = maskLabels
        this.maskImages = maskImages
        this.minimumAlpha = minimumAlpha
    }
}

public class LDSessionReplayOptions {
    @JvmField var isEnabled: Boolean = true
    @JvmField var serviceName: String = ""
    @JvmField var privacy: LDPrivacyOptions = LDPrivacyOptions()

    constructor()

    constructor(
        isEnabled: Boolean,
        serviceName: String,
        privacy: LDPrivacyOptions
    ) {
        this.isEnabled = isEnabled
        this.serviceName = serviceName
        this.privacy = privacy
    }
}

File: sdk/@launchdarkly/mobile-dotnet/macios/native/LDObserve/Sources/OptionsBridge.swift
import Foundation

@objc(ObjcObservabilityOptions)
public final class ObjcObservabilityOptions: NSObject {
    @objc public var serviceName: String = ""
    @objc public var serviceVersion: String = ""
    @objc public var otlpEndpoint: String = ""
    @objc public var backendUrl: String = ""
    @objc public var attributes: NSDictionary?
    // Not consumed yet — network tracing is handled on the .NET side. Kept for future iOS-native URL session instrumentation.
    @objc public var networkRequests: Bool = true
    @objc public var launchTimes: Bool = true

    @objc public override init() {
        super.init()
    }
}

@objc(ObjcSessionReplayOptions)
public final class ObjcSessionReplayOptions: NSObject {
    @objc public var isEnabled: Bool = true
    @objc public var maskTextInputs: Bool = true
    @objc public var maskWebViews: Bool = false
    @objc public var maskLabels: Bool = false
    @objc public var maskImages: Bool = false

    @objc public override init() {
        super.init()
    }
}

@objc(ObjcEnvironmentMetadata)
public final class ObjcEnvironmentMetadata: NSObject {
    @objc public var credential: String = ""
    @objc public var sdkName: String = ""
    @objc public var sdkVersion: String = ""
    @objc public var applicationId: String = ""
    @objc public var applicationVersion: String = ""

    @objc public override init() {
        super.init()
    }
}
